### PR TITLE
adding message.data_event so that topics can have multiple purposes

### PR DESCRIPTION
**Pull request by perryqh**

### CODE REVIEW

File: lib/g5_pub_sub/base_subscriber.rb [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
       raise 'must override!'
     end
 
+    def handle_message(message)
+      return on_received_message(message) if message.data_event.blank?
+
+      # if subscribers have implemented an event specific method, delegate to it
+      event_specific_method = "on_received_#{message.data_event}_message"
+      return send(event_specific_method, message) if respond_to?(event_specific_method)
+
```
Comment (perryqh) [inline lib/g5_pub_sub/base_subscriber.rb:18]: This enables a subscriber to do things like:
```
class FormSubscriber < G5PubSub::BaseSubscriber
  pub_sub_name :forms

  def on_received_created_message(message)
      # create the form in the message
  end

  def on _received_deleted_message(message)
       # delete the associated form
  end 
end
```

Comment (mmitchellg5) [reply lib/g5_pub_sub/base_subscriber.rb:18]: does an event _have_ to fall into one of these types?

Comment (perryqh) [reply lib/g5_pub_sub/base_subscriber.rb:18]: I'll update the changelog. Right now, it needs to be (nil, created, updated, or deleted).
It's possible that this won't be sufficient in the future. We can either not enforce 'inclusion' or add events as needed in the future. I went with validating 'inclusion'. I don't have a strong opinion about it.

Comment (mmitchellg5) [reply lib/g5_pub_sub/base_subscriber.rb:18]: my only thought is we wouldnt want to cut ourselves off at only having "data" type events, ie I'm thinking events could be more "behavior" type like 'this thing happened'  you decide what that means to you

Comment (perryqh) [reply lib/g5_pub_sub/base_subscriber.rb:18]: I'll remove the validation

**Pull request merged**